vault/resource_secrets_sync_gcp_destination_test.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	"github.com/hashicorp/terraform-provider-vault/testutil"
+)
+
+func TestGCPSecretsSyncDestination(t *testing.T) {
+	destName := acctest.RandomWithPrefix("tf-sync-dest-gcp")
+
+	resourceName := "vault_secrets_sync_gcp_destination.test"
+
+	credentials, _ := testutil.GetTestGCPCreds(t)
+
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: providerFactories,
+		PreCheck: func() {
+			testutil.TestAccPreCheck(t)
+			SkipIfAPIVersionLT(t, testProvider.Meta(), provider.VaultVersion116)
+		}, PreventPostDestroyRefresh: true,
+		Steps: []resource.TestStep{
+			{
+				Config: testGCPSecretsSyncDestinationConfig(credentials, destName, defaultSecretsSyncTemplate),
+				Check: resource.ComposeTestCheckFunc(
+
+					resource.TestCheckResourceAttr(resourceName, consts.FieldName, destName),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldCredentials, credentials),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldSecretNameTemplate, defaultSecretsSyncTemplate),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldType, gcpSyncType),
+					resource.TestCheckResourceAttr(resourceName, "custom_tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "custom_tags.foo", "bar"),
+				),
+			},
+			testutil.GetImportTestStep(resourceName, false, nil,
+				consts.FieldCredentials),
+		},
+	})
+}
+
+func testGCPSecretsSyncDestinationConfig(credentials, destName, templ string) string {
+	ret := fmt.Sprintf(`
+resource "vault_secrets_sync_gcp_destination" "test" {
+  name                 = "%s"
+  credentials          = <<CREDS
```

Comment:
is this how we do file paths?

Reply:
I believe this is [heredoc](https://phoenixnap.com/kb/bash-heredoc#:~:text=A%20HereDoc%20is%20a%20multiline,neater%20and%20easier%20to%20understand.) syntax to set the file's content inline.

To use a file reference, a user would use the Terraform built-in file() method like `file("path/to/gcp/credentials.json")`.

Reply:
Max is correct! In my manual TF config files, I was passing in a file path and using the `file` helper to parse the JSON credentials as Max showed above. However, I also wanted to make sure that passing JSON credentials directly also works, so I opted to use heredoc syntax within the test config to ensure that users can provide credentials through these different ways :)